### PR TITLE
Additional test

### PR DESCRIPTION
**Pull request by WLefever-Cegeka**

### CODE REVIEW

File: ldes-fragmentisers/ldes-fragmentisers-substring/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/substring/SubstringFragmentationStrategyWrapper.java [hunk 1]
```diff
@@ -20,18 +22,21 @@ public class SubstringFragmentationStrategyWrapper implements FragmentationStrat
 	public FragmentationStrategy wrapFragmentationStrategy(ApplicationContext applicationContext,
```
Comment (sverholen) [inline ldes-fragmentisers/ldes-fragmentisers-substring/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/substring/SubstringFragmentationStrategyWrapper.java:22]: Why use the ApplicationContext here ?

File: ldes-fragmentisers/ldes-fragmentisers-timebased/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/timebased/services/OpenFragmentProvider.java [hunk 2]
```diff
@@ -2,29 +2,43 @@
...
 	public OpenFragmentProvider(TimeBasedFragmentCreator fragmentCreator,
-			LdesFragmentRepository ldesFragmentRepository) {
+			LdesFragmentRepository ldesFragmentRepository, TimebasedFragmentationConfig timebasedFragmentationConfig) {
 		this.fragmentCreator = fragmentCreator;
 		this.ldesFragmentRepository = ldesFragmentRepository;
+		this.timebasedFragmentationConfig = timebasedFragmentationConfig;
 	}
 
-	public LdesFragment retrieveOpenFragmentOrCreateNewFragment(LdesFragment parentFragment) {
+	public Pair<LdesFragment, Boolean> retrieveOpenFragmentOrCreateNewFragment(LdesFragment parentFragment) {
```
Comment (sverholen) [inline ldes-fragmentisers/ldes-fragmentisers-timebased/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/timebased/services/OpenFragmentProvider.java:22]: Good fix to use a Pair here, but it's completely unclear what the boolean means unless you visit this class. Maybe abstract this in a class ?

File: ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/config/ExecutorsConfig.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+@Configuration
+public class ExecutorsConfig {
+
+	@Bean
+	public ExecutorService executorService() {
+		return Executors.newCachedThreadPool();
```
Comment (sverholen) [inline ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/config/ExecutorsConfig.java:14]: Don't know how an Executor works, but i'd expect a pool size here as an extra (minor) performance improvement.

**Review by sverholen — APPROVED** [review]

**Review by Yalz — APPROVED** [review]
In general, looks like a good job ! 
Some post merge actions:

- Put relations inside of Fragments again (access only through mongotemplate push/pull operations)
- TreeNodeRemoverImpl: replace save with mongoTemplate operation
- replace member saving (which is now exists + upsert) with insert (with duplicateKeyException handling)

**Comment by Yalz** [comment]
General comment.. I think we might be overusing the upsert functionality.. 
An upsert will update a record or create it if not yet existant.. However.. If a record does not exist, this should ring alarm bells. 
Otherwise we risk creating partial records

File: ldes-fragmentisers/ldes-fragmentisers-substring/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/substring/SubstringFragmentationStrategyWrapper.java (hunk 1 above)
Comment (WLefever-Cegeka) [reply ldes-fragmentisers/ldes-fragmentisers-substring/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/substring/SubstringFragmentationStrategyWrapper.java:22]: ApplicationContext is part of the general interface of `FragmentationStrategyWrapper`:
`FragmentationStrategy wrapFragmentationStrategy(ApplicationContext applicationContext,
			FragmentationStrategy fragmentationStrategy, ConfigProperties fragmentationProperties);`
Initial the idea was that each fragmentationstrategy would be able to get certain beans they need out from the ApplicationContext, e.g. for TimeBasedFragmentation it was needed to retrieve members to determine the max value of `generatedAtTime`, so TimeBasedFragmentation could get the MemberRepository bean.

Maybe to revise with the team as now all FragmentationStrategies use the same beans so maybe it's not very relevant anymore.

File: ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/config/ExecutorsConfig.java (hunk 3 above)
Comment (WLefever-Cegeka) [reply ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/config/ExecutorsConfig.java:14]: Thought about it. Maybe something for in config?

**Pull request merged**